collector/logs/sources/journal/cursor.go

```diff
@@ -0,0 +1,68 @@
+package journal
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/adx-mon/pkg/logger"
+	"github.com/cespare/xxhash"
+)
+
+type journalcursor struct {
+	Cursor string `json:"cursor"`
+}
+
+func cursorPath(cursorDirectory string, filters []string, database string, table string) string {
+	hasher := xxhash.New()
+	// filters are order dependent, so do not sort before creating the hash.
+	for _, filter := range filters {
+		hasher.Write([]byte(filter))
+	}
+	filterHash := hasher.Sum64()
+	fileName := fmt.Sprintf("journal_%s_%s_%x.cursor", database, table, filterHash)
+	sanitized := strings.ReplaceAll(fileName, string(filepath.Separator), "_")
```

Comment:
```suggestion
	sanitized := filepath.Clean(fileName)
```

Might be a safter way to do the same thing

Reply:
Sadly `Clean` doesn't quite do this despite the name - it just removes duplicate separators and simplifies `..` and `.` bits within the path, but if there are separators in the path it keeps em.